src/main/java/seedu/duke/data/notebook/Note.java

```diff
@@ -99,4 +105,20 @@ public String getTagsName() {
     public void setTags(ArrayList<Tag> tags) {
         this.tags = tags;
     }
+
+    public String toSaveString() {
+        String noteDetails = "";
+        String tagDetails = "";
+
+        for (Tag tag: this.tags) {
+            tagDetails += PREFIX_DELIMITER + PREFIX_TAG + " " + tag.toSaveString() + " ";
+        }
+        noteDetails += PREFIX_DELIMITER + PREFIX_TITLE + " " + this.title + " "
+                    + PREFIX_DELIMITER + PREFIX_PIN + " " + this.isPinned + " "
+                    //+ PREFIX_DELIMITER + PREFIX_ARCHIVE + " " + this.isArchived + " "
```

Comment:
@R-Ramana I think the boolean is still needed.

Reply:
@prachi2023 FYI, have re-added it in the Note class. Please take a look if you need to use.